addons/l10n_in/models/company.py

```diff
@@ -84,3 +84,17 @@ def _check_l10n_in_pan(self):
     def action_update_state_as_per_gstin(self):
         self.ensure_one()
         self.partner_id.action_update_state_as_per_gstin()
+
+    def _get_token_list(self):
+        return []
+
+    @api.onchange('vat')
+    def _onchange_vat(self):
+        if self.account_fiscal_country_id.code == 'IN':
+            if token_list := self._get_token_list():
```

Comment:
```suggestion
        if self.account_fiscal_country_id.code == 'IN' and (token_list := self._get_token_list()):
```
Can be done in one line